contracts/Token.sol

```diff
@@ -1,68 +1,118 @@
+import "@openzeppelin/contracts-upgradeable/proxy/utils/UUPSUpgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/token/ERC20/ERC20Upgradeable.sol";
+import "./IVestingToken.sol";
 
-contract Token is Ownable, ERC20, ICount {
+contract Token is Initializable, OwnableUpgradeable, ERC20Upgradeable, UUPSUpgradeable, IVestingToken {
     using SafeMath for uint;
 
-    address immutable private _existing_holders;
-    address immutable private _outlier_ventures;
-    address immutable private _investors;
-    address immutable private _foundation;
-    address immutable private _growth;
-    address immutable private _advisors;
-    address immutable private _curve;
+    address public constant _existingHolders = 0x0364eAA7C884cb5495013804275120ab023619A5;
+    address public constant _outlierVentures = 0x0364eAA7C884cb5495013804275120ab023619A5;
+    address public constant _investors = 0x0364eAA7C884cb5495013804275120ab023619A5;
+    address public constant _foundation = 0xB1C0a6ea0c0E54c4150ffA3e984b057d25d8b28C;
+    address public constant _growth = 0x0364eAA7C884cb5495013804275120ab023619A5;
+    address public constant _advisors = 0x0364eAA7C884cb5495013804275120ab023619A5;
+    address public constant _curve = 0x0364eAA7C884cb5495013804275120ab023619A5;
 
-    uint256 private _holderCount;
-    uint256 immutable private _startBlock;
+    uint256 public holderCount;
+    address public vestingContract;
 
-    function Count() external override view returns (uint256) {
-        return _holderCount;
-    }
+    /**
+     * @dev Our constructor (with UUPS upgrades we need to use initialize(), but this is only
+     *      able to be called once because of the initializer modifier.
+     */
+    function initialize() public initializer {
+        __Ownable_init();
+        __ERC20_init("moda", "MODA");
 
-    constructor() ERC20("moda", "MODA") {
-        _existing_holders = 0x0364eAA7C884cb5495013804275120ab023619A5;
-        _outlier_ventures = 0x0364eAA7C884cb5495013804275120ab023619A5;
-        _investors = 0x0364eAA7C884cb5495013804275120ab023619A5;
-        _foundation = 0xB1C0a6ea0c0E54c4150ffA3e984b057d25d8b28C;
-        _growth = 0x0364eAA7C884cb5495013804275120ab023619A5;
-        _advisors = 0x0364eAA7C884cb5495013804275120ab023619A5;
-        _curve = 0x0364eAA7C884cb5495013804275120ab023619A5;
-
-        mintWithCount(0x0364eAA7C884cb5495013804275120ab023619A5, 2000000 * 10 ** 18);
-        mintWithCount(0x0364eAA7C884cb5495013804275120ab023619A5, 300000 * 10 ** 18);
-        mintWithCount(0x0364eAA7C884cb5495013804275120ab023619A5, 500000 * 10 ** 18);
-        mintWithCount(0xB1C0a6ea0c0E54c4150ffA3e984b057d25d8b28C, 3500000 * 10 ** 18);
-        mintWithCount(0x0364eAA7C884cb5495013804275120ab023619A5, 1000000 * 10 ** 18);
-        mintWithCount(0x0364eAA7C884cb5495013804275120ab023619A5, 1200000 * 10 ** 18);
-        mintWithCount(0x0364eAA7C884cb5495013804275120ab023619A5, 1500000 * 10 ** 18);
```

Comment:
I assume we would add via a deploy script once these addresses are known, hence why they're in the constructor.

Reply:
I'll make it an `initialize` parameter so we can fiddle these when we deploy.

Reply:
Cool, then ill merge